gradle/releaser.gradle

```diff
@@ -124,19 +124,19 @@ if (rootProject.hasProperty("artifactory_publish_password")) {
 }
 
 static def repoKeyFor(p) {
-	def repos = [
-			// New scheme
-			(~/^\d+\.\d+\.\d+$/) : "libs-release-local",
-			(~/^\d+\.\d+\.\d+-SNAPSHOT$/) : "libs-snapshot-local",
-			(~/^\d+\.\d+\.\d+-M\d+$/) : "libs-milestone-local",
-			(~/^\d+\.\d+\.\d+-RC\d+$/) : "libs-milestone-local",
-			// Old scheme
-			(~/^\d+\.\d+\.\d+\.RELEASE$/) : "libs-release-local",
-			(~/^\d+\.\d+\.\d+\.BUILD-SNAPSHOT$/) : "libs-snapshot-local",
-			(~/^\d+\.\d+\.\d+\.M\d+$/) : "libs-milestone-local",
-			(~/^\d+\.\d+\.\d+\.RC\d+$/) : "libs-milestone-local",
-			// Test
-			(~/^\d+\.\d+\.\d+-.+-SNAPSHOT$/) : "libs-snapshot-local",
-	]
-	return repos.find {e -> p.version =~ e.key}.value
+	switch (p.version) {
+		case (~/^\d+\.\d+\.\d+$/):
+		case (~/^\d+\.\d+\.\d+\.RELEASE$/):
+			return "libs-release-local";
+		case (~/^\d+\.\d+\.\d+-SNAPSHOT$/):
+		case (~/^\d+\.\d+\.\d+\.BUILD-SNAPSHOT$/):
+			return "libs-snapshot-local";
+		case (~/^\d+\.\d+\.\d+-M\d+$/):
+		case (~/^\d+\.\d+\.\d+-RC\d+$/):
+		case (~/^\d+\.\d+\.\d+\.M\d+$/):
+		case (~/^\d+\.\d+\.\d+\.RC\d+$/):
```

Comment:
```suggestion
		case (~/^\d+\.\d+\.\d+(-|\.)M\d+$/):
		case (~/^\d+\.\d+\.\d+(-|\.)RC\d+$/):
```

Reply:
it took me some time to figure out the difference (`-` vs `.`), hence the suggestion to use `(X|Y)` :)

Reply:
I hesitated to add back comments which explain that this supports old vs new scheme. In that particular case, that may be the best way to make the difference stand out, wdyt?

Reply:
I got an idea: add an example next to every pattern:

```groovy
		case (~/^\d+\.\d+\.\d+-M\d+$/):   // 1.2.3-M1
		case (~/^\d+\.\d+\.\d+-RC\d+$/):  // 1.2.3-RC1
		case (~/^\d+\.\d+\.\d+\.M\d+$/):  // 1.2.3.M1
		case (~/^\d+\.\d+\.\d+\.RC\d+$/): // 1.2.3.RC1
```